overlord/restart/restart.go

```diff
@@ -394,13 +395,27 @@ func Pending(st *state.State) (bool, RestartType) {
 		return false, RestartUnset
 	}
 	rm := cached.(*RestartManager)
-	return rm.restarting != RestartUnset, rm.restarting
+	return rm.Pending()
+}
+
+// Pending returns whether a restart was requested with Request and of which type.
+// NOTE: the state does not need to be locked to fetch this information.
+func (rm *RestartManager) Pending() (bool, RestartType) {
+	if rm == nil {
+		return false, RestartUnset
+	}
```

Comment:
this is odd - why?

Reply:
pebble doesn't have this bit so a bit unsure? do we have tests were RestartManager is retrieved from somewhere and is nil?